test/new-e2e/pkg/environments/aws/host/host.go

```diff
@@ -267,3 +267,43 @@ func Provisioner(opts ...ProvisionerOption) e2e.TypedProvisioner[environments.Ho
 
 	return provisioner
 }
+
+// AwsHostParams is the set of properties for the AWS Host environment.
+type AwsHostParams struct {
+	// Name is the name of the EC2 VM.
+	Name string
+	// PulumiContext is the Pulumi context to use.
+	PulumiContext *pulumi.Context
+	// AwsEnvironment is the AWS environment to use.
+	AwsEnvironment *aws.Environment
+	// Options is the set of options for the EC2 VM.
+	Options []ec2.VMOption
+	// Importer is the component that will be used to import the EC2 host from pulumi.
+	Importer *remote.HostOutput
+}
+
+func NewHost(params AwsHostParams) (host *remote.Host, err error) {
```

Comment:
This kind of function we can reuse everywhere :+1: :)

Reply:
🙌